src/pages/ImportPage.tsx

```diff
@@ -1,37 +1,148 @@
+import {
+  Button,
+  Heading,
+  HStack,
+  Input,
+  Stack,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
 import { Project } from "@microbit/makecode-embed/react";
-import { useEffect } from "react";
-import { IntlShape, useIntl } from "react-intl";
+import { ReactNode, useCallback, useEffect, useRef, useState } from "react";
+import { FormattedMessage, IntlShape, useIntl } from "react-intl";
 import { useNavigate } from "react-router";
 import { useSearchParams } from "react-router-dom";
+import DefaultPageLayout, {
+  HomeMenuItem,
+  HomeToolbarItem,
+} from "../components/DefaultPageLayout";
 import { useDeployment } from "../deployment";
+import { useProject } from "../hooks/project-hooks";
 import { MicrobitOrgResource } from "../model";
 import { useStore } from "../store";
-import { createDataSamplesPageUrl } from "../urls";
+import { createDataSamplesPageUrl, createHomePageUrl } from "../urls";
+import { useConnectionStage } from "../connection-stage-hooks";
+import { defaultProjectName, validateProjectName } from "../project-name";
 
 const ImportPage = () => {
-  const navigate = useNavigate();
   const intl = useIntl();
+  const navigate = useNavigate();
   const { activitiesBaseUrl } = useDeployment();
   const resource = useMicrobitResourceSearchParams();
-  const loadProject = useStore((s) => s.loadProject);
+  const project = useRef<Project>();
+  const [name, setName] = useState<string>(defaultProjectName);
+  const isValidSetup = validateProjectName(name);
 
   useEffect(() => {
     const updateAsync = async () => {
       if (!resource || !activitiesBaseUrl) {
         return;
       }
-      const code = await fetchMicrobitOrgResourceTargetCode(
+      project.current = await fetchMicrobitOrgResourceProjectCode(
```

Comment:
I think we should discuss whether we should have a loading state here rather than what's going to be UI that does something different depending on the delay in the fetch.

Reply:
Button loading state added here [5c50681](https://github.com/microbit-foundation/ml-trainer/pull/394/commits/5c506817f9858975034fb4205a08f4eecfff56cb)